### DISCUSSION TITLE
Live Stream Mixer

### DISCUSSION THREAD

**Discussion by CabbageMessiah** [brunoherbelin/vimix#70] [Ideas]
Hello!

I wondered if VJ software could serve as a creative tool for live streams. I started toying around with Vimix on this hunch and was really happy to see all the effects and ways to manipulate sources that seem to be what I was looking for. Using OBS Studio, I can capture the "Output Video Live Mixer" window, but was unable to capture the SRT Broadcast option.

Right now, my ideas are limited to creating visuals while playing music in an other application, and I saw this was the convention recommended in the documentation. Are there any other applications known to complement Vimix, though? That use OSC and can trigger actions synchronously?

If there's an easy way to get a direct capture of the output in OBS, it'd be great, but that makes me imagine if this had the capacity to broadcast to a streaming service, like OBS Studio does, it'd be an awesome way to utilize its functions. I could just throw in a video source or game source, and pull the sources into focus or make them play together and mix the presentation of a broadcast.

I'm glad I found Vimix anyway and wanted to compliment it. I'm looking forward to its development. Thanks Bruno and Co.

**Comment by brunoherbelin** [comment]
Thanks for the comment!

Yes, vimix is definitely a good companion for live streaming with e.g. OBS. The screenshot bellow shows different possibilities, with window capture of with V4L2loopback :

![Screenshot from 2023-03-22 23-44-18](https://user-images.githubusercontent.com/60393189/227055367-836e3911-1559-4afc-a8cc-c383fc5c899d.png)

I am also looking at ways to improve this. I investigated with shared memory from gstreamer but this is not supported by OBS.  Latest OBS seems to supports `pipewire` inputs, which could be a possible output for vimix, maybe.. On MacOS there is Syphon, but its not really my priority. Alternatively, I could propose an OBS plugin: maybe I will investigate this.

Regarding the broadcasting of output direclty from vimix, I would be ok for a 100% open source approach (but I am not sure Twitch or other services would not link to a monetization in some way, which would be a no go for me). In short : I am not very much into this and would welcome contributions and suggestions.
